pkg/plugin/plugin_test.go

```diff
@@ -256,3 +257,30 @@ func TestRecoverEtcdCluster(t *testing.T) {
 		t.Errorf("plugin.RecoverEtcdCluster error = %v", err)
 	}
 }
+
+func TestRotateClusterSecrets(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	mockGen := mock_config.NewMockGenerator(mockCtrl)
+	mockUp := mock_cluster.NewMockUpgrader(mockCtrl)
+	mockArm := mock_arm.NewMockGenerator(mockCtrl)
+
+	mockGen.EXPECT().InvalidateSecrets(nil).Return(nil)
```

Comment:
How do these tests work?  Looks like this and the following are doing `nil` checks.  Just for my understanding.

Reply:
@kwoodson yes those look weird but those calls are setting up expectations and validating the order of calls in the tested function/method. I could also have done `mockGen.EXPECT().InvalidateSecrets(kenny).Return(nil)` and in this case, if the tested function is passed `kenny`, then I would expect that InvalidateSecrets(kenny) is called internally within that function and returns no error (nil).

I can have a call with you sometime to explain more about gomock